include/openthread/ping_sender.h

```diff
@@ -58,29 +58,57 @@ typedef struct otPingSenderReply
     uint8_t      mHopLimit;       ///< Hop limit.
 } otPingSenderReply;
 
+/**
+ * This structure represents statistics of several ping requests.
+ *
+ */
+typedef struct otPingSenderStatistics
+{
+    uint32_t mTotalPingCount;     ///< The number of total ping requests to send.
+    uint32_t mSentCount;          ///< The number of ping requests already sent.
+    uint32_t mReceivedCount;      ///< The number of ping replies received.
+    uint32_t mTotalRoundTripTime; ///< The total round trip time of ping requests.
+    uint32_t mMinRoundTripTime;   ///< The min round trip time among ping requests.
+    uint32_t mMaxRoundTripTime;   ///< The max round trip time among ping requests.
```

Comment:
I used `uint32_t` because `otPingSenderReply` is using this type.

Also, I think `mTotalRoundTripTime` could be large (more than 100 seconds) when we use echo a lot of times in a ping request.

Reply:
> Also, I think `mTotalRoundTripTime` could be large (more than 100 seconds) when we use echo a lot of times in a ping request.

We can save the average RTT and update average RTT for each new reply?
```
NewAvgRtt = ((OldAvgRtt * Count) + Reply.RTT) / (Count+1)
```

Reply:
> I used `uint32_t` because `otPingSenderReply` is using this type.

Should we change `mRoundTripTime` of `otPingSenderReply` to `uint16_t` ?
I think 65535 ms is enough for storing RoundTripTime.

I realized that `otPingSenderReply` is allocated in stack so its memory size is less restricted. 
But should make `otPingSenderStatistics` use less memory and consistent with `otPingSenderReply`. 

Thoughts? 
@abtink

Reply:
I'm inclined to keep the variable mTotalRoundTripTime, and make it a uint32_t. 
I tried calculating the average RTT using the equation above, and found the precision poor.

Reply:
One thing to consider related to RTT is that if we may deal with SED devices which poll very infrequently and then we may have rather long ping delay. `uint16_t` would give us ~65sec. I think it may be possible that we have SEDs that data poll in order of minutes.  So if we can keep it as `uint32_t` it may be better.

> We can save the average RTT and update average RTT for each new reply?
> I tried calculating the average RTT using the equation above, and found the precision poor.

I think the sum model is fine here. But if we want to use running average model, we may need to cauclate the average of "values times some factor" to increase precision. As an example `RssAverager` in `link_quality.hpp` keeps track of average of "RSS times 8".

Reply:
I don't have strong opinion against `uint32_t mTotalRoundTripTime`, even though I think Thread users usually do no care about ping delay precision under 1ms. 

> I think it may be possible that we have SEDs that data poll in order of minutes. So if we can keep it as uint32_t it may be better.

Agree SED ping delay can be larger than 65s. 

> But if we want to use running average model, we may need to cauclate the average of "values times some factor" to increase precision.

That would make reduce the range of uint16_t from 65s to less than 10s, which would not be enough. I would prefer using uint32_t to this proposal.